test/Microsoft.ML.Benchmarks/StochasticDualCoordinateAscentClassifierBench.cs

```diff
@@ -67,6 +54,27 @@ public void Setup()
             }
         }
 
+        [Benchmark]
+        public PredictionModel<IrisData, IrisPrediction> TrainIris() => TrainCore();
+
+        [Benchmark]
+        public float[] PredictIris() => s_trainedModel.Predict(s_example).PredictedLabels;
+
+        [Benchmark]
+        public void PredictIrisBatchOf1() => Consume(s_trainedModel.Predict(s_batches[0]));
```

Comment:
Would it be possible for BDN to fix this kind of issue in the future?  Possibly by enumerating the IEnumerable itself?  Or maybe even writing a code analyzer to catch these kinds of mistakes?

Reply:
I started discussion about that https://github.com/dotnet/BenchmarkDotNet/issues/858 Personally I think we should give an error + expose a public IEnumeralbe.Consume extension method that will do it for the people. So the benchmark would be:

```cs
public void PredictIrisBatchOf1() => s_trainedModel.Predict(s_batches[0]).Consume();
```

Reply:
That sounds good to me. Thanks for fixing this, and suggesting how to make this better in the future for other BDN users.